### PR TITLE
Add sequence erase op

### LINKED ISSUE #7171: Add sequence erase operator
Remove the specific tokens in sequence data, and modify the LoD information accordingly.

### PR DESCRIPTION
**Pull request by kuke**
Resolve #7171

### CODE REVIEW

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    return out_seq, new_lod0
+
+
+class TestSequenceEraseOp(OpTest):
+    def setUp(self):
+        self.op_type = "sequence_erase"
+        in_seq = np.random.randint(0, 10, (10, 1)).astype("int32")
+        lod = [[0, 3, 6, 10]]
+        tokens = [2, 3, 5]
+        out_seq, new_lod0 = sequence_erase(in_seq, lod[0], tokens)
```
Comment (wanghaoshuang) [inline python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: Are you sure it is lod[0] but not lod[-1]?

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    tokens = [2, 5]
+    out_seq, new_lod = sequence_erase(in_seq, lod0, tokens)
+    
+    print lod0, new_lod
+    print("compare")
+    for i in range(0,  len(lod0)-1):
+        print(np.transpose(in_seq[lod0[i] : lod0[i+1]]))
+        print(np.transpose(out_seq[new_lod[i] : new_lod[i+1]]))
+        print("\n")
+    """
```
Comment (wanghaoshuang) [inline python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: Delete line44 - line56?

File: paddle/operators/sequence_erase_op.cc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    AddOutput("Out",
+              "(2-D LoDTensor with the 2nd dim. equal to 1) "
+              "Output LoDTensor of SequenceEraseOp.");
+    AddAttr<std::vector<int>>("tokens",
+                              "(vector<int>) Tokens need to be erased from "
+                              "input sequences.");
+    AddComment(R"DOC(
+Sequence Erase Operator.
+
+Sequence erase operator erases tokens specified by Attr(tokens) in the input
```
Comment (wanghaoshuang) [inline paddle/operators/sequence_erase_op.cc]: `in the input` -> `from the input `

File: paddle/operators/sequence_erase_op.cc [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                              "(vector<int>) Tokens need to be erased from "
+                              "input sequences.");
+    AddComment(R"DOC(
+Sequence Erase Operator.
+
+Sequence erase operator erases tokens specified by Attr(tokens) in the input 
+sequences Input(X), and outputs the remaining data and modifies the LoD 
+information at the same time. For example, given a 2-D LoDTensor
+
+    X = [[2, 2, 6, 1, 3, 9, 6, 1, 0, 1]]^T
```
Comment (wanghaoshuang) [inline paddle/operators/sequence_erase_op.cc]: 这种写法不对吧？
X.dims=[1, 10] ?

File: paddle/operators/sequence_erase_op.h [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    auto* in = ctx.Input<framework::LoDTensor>("X");
+    auto* out = ctx.Output<framework::LoDTensor>("Out");
+
+    auto lod = in->lod();
+    PADDLE_ENFORCE_EQ(lod.size(), 1UL, "Only support one level sequence now.");
+    auto tokens = ctx.Attr<std::vector<int>>("tokens");
+    auto in_len = in->numel();
+    auto in_dat = in->data<T>();
+    auto lod0 = lod[0];
```
Comment (wanghaoshuang) [inline paddle/operators/sequence_erase_op.h]: Are you sure it is lod[0] but not lod[-1]?

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import unittest
+import numpy as np
+from op_test import OpTest
+
+
+def sequence_erase(in_seq, lod0, tokens):
+    # num_erased[i]: the number of elments to be removed before #i elements
+    num_erased = [0] * (len(in_seq) + 1)
```
Comment (wanghaoshuang) [inline python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: 不用浪费空间记录`num_erased `吧？按`lod0`扫一遍`input`, 符合条件的copy到`output`, 同时记录下`new_lod0`.
`
for i in range(len(lod0)):
    count=0
    for tokens_index in range(i, i+1):
        if in[tokens_index] not in tokens:
            count++
            out.append(in[tokens_index])
        new_lod[i] = count
   
`

C++端代码同理

File: paddle/operators/sequence_erase_op.cc (hunk 3 above)
Comment (kuke) [reply paddle/operators/sequence_erase_op.cc]: Done

File: paddle/operators/sequence_erase_op.cc (hunk 4 above)
Comment (kuke) [reply paddle/operators/sequence_erase_op.cc]: 按照约定，X.dims应该为[10, 1]

File: paddle/operators/sequence_erase_op.h (hunk 5 above)
Comment (kuke) [reply paddle/operators/sequence_erase_op.h]: Please note line 30
```
PADDLE_ENFORCE_EQ(lod.size(), 1UL, "Only support one level sequence now.");
```

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import unittest
+import numpy as np
+from op_test import OpTest
+
+
+def sequence_erase(in_seq, lod0, tokens):
+    # num_erased[i]: the number of elments to be removed before #i elements
```
[the remark this answers is not in the document]
Comment (kuke) [reply python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: Done

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py (hunk 6 above)
Comment (kuke) [reply python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: 用两个循环最主要的原因是在C++端为Output(Out)分配内存时要知道其实际的size；test则不受此限，已将两个循环合并为一个

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+            out_seq[i - num_erased[i]] = in_seq[i]
+        # else in_seq[i] needs to be removed 
+    return out_seq, new_lod0
+
+
+class TestSequenceEraseOp(OpTest):
+    def setUp(self):
+        self.op_type = "sequence_erase"
+        in_seq = np.random.randint(0, 10, (10, 1)).astype("int32")
+        lod = [[0, 3, 6, 10]]
```
[the remark this answers is not in the document]
Comment (kuke) [reply python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: `n > 1` is illegal;

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py (hunk 1 above)
Comment (kuke) [reply python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: Only one-level LoD

File: python/paddle/v2/fluid/tests/test_sequence_erase_op.py (hunk 2 above)
Comment (kuke) [reply python/paddle/v2/fluid/tests/test_sequence_erase_op.py]: Done

**Pull request merged**